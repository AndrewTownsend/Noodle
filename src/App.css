#error-page {
  margin:auto;
  width: 30%;
}

#addnewrow input[type=checkbox]{
display:none;
}

label.checkboxlabel {
  display: inline-block;
  width:100%;
  height:100%;
  background-color: var(--bs-secondary);
}

input[type=checkbox]:checked + .checkboxlabel{
  background-color: var(--bs-primary);
  }

  #submitButtonContainer {
    width:100%;
    margin:auto;
    display: flex;
    justify-content: center;
  }

  .goingPrimary {
    background-color: var(--bs-primary);
  }