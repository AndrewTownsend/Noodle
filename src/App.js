import "./App.css";
import React, { useState } from "react";
import { Create } from "./Create";
import { Alert } from "react-bootstrap";

function App() {
  const [errorMessage, setErrorMessage] = useState("");
  const [successMessage, setSuccessMessage] = useState("");

  return (
    <div className="App">
      <div className="header">
        <h1>Noodle</h1>
        Scheduling events should be easy. That's using your Noodle.
      </div>
      {errorMessage && (
        <Alert variant={"info"}>
          It may seem impastable, but something's gone wrong.
          <p>{errorMessage}</p>
        </Alert>
      )}
      {successMessage && <Alert variant={"success"}>{successMessage}</Alert>}
      <Create
        setErrorMessage={setErrorMessage}
        setSuccessMessage={setSuccessMessage}
      />
    </div>
  );
}

export default App;
