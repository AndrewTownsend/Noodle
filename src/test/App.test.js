import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "../App";
import { MemoryRouter } from "react-router-dom";

test("renders", () => {
  render(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
  const linkElement = screen.getByText(/using your noodle/i);
  expect(linkElement).toBeInTheDocument();
});

test("renders and looks the same as before", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("doesn't let you submit form until required fields are filled out", async () => {
  //this will never work until the calendar is a11y, so we'll get to it later
  //   render(
  //     <RenderRouteWithOutletContext context={context}>
  //       <Create />
  //     </RenderRouteWithOutletContext>
  //   );
  //   const submitButton = await screen.findByText("Submit");
  //   expect(submitButton).toBeDisabled();
  //   userEvent.click(
  //     screen.getByRole("textbox", { name: /What's your event called?/ })
  //   );
  //   userEvent.type("asdf;");
  //   userEvent.click(screen.getByRole("textbox", { name: /What's your name?/ }));
  //   userEvent.type("asdfjk");
  //   const calendar = screen.getByRole("dialog");
  //   //eslint-disable-next-line
  //   const firstDayOfMonth = calendar.querySelectorAll(".rdmp-day .sd")[0]; //this is an ugly hack and it would be better to switch to an accessible calendar picker or update the RMDP picker to be accessible
  //   screen.debug(firstDayOfMonth);
  //   userEvent.click(firstDayOfMonth);
  //   //expect(submitButton).toBeEnabled();
});
