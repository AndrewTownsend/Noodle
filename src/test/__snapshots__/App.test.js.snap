// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and looks the same as before 1`] = `
<div
  className="App"
>
  <header>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="border-bottom col"
        >
          <p
            className="h1 py-3"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Noodle
            </a>
          </p>
          <p>
            Scheduling events should be easy. That's using your Noodle.
          </p>
        </div>
      </div>
    </div>
  </header>
  <main>
    <div
      className="py-3 container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        />
      </div>
    </div>
  </main>
  <footer>
    <div
      className="py-3 container"
    >
      <div
        className="row"
      >
        <div
          className="col"
        >
          <p>
            Â© 2023
          </p>
        </div>
        <div
          className="col-sm-auto"
        >
          <a
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </div>
        <div
          className="col-sm-auto"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
